modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/ballooneditor/plugin.js

```diff
@@ -307,6 +307,40 @@
 
 				originalContextManagerCheck.call(this, selection);
 			};
+
+			const ckeditorWindow = CKEDITOR.document.getWindow();
+
+			ckeditorWindow.on('click', (event) => {
+				const target = event.data.getTarget();
```

Comment:
@markocikos the changes mostly LGTM, I just had a doubt (and I haven't been able to check, so bear with me), in "ckeditor" event listeners, doesn't the `event` argument have an `editor` property which is a reference to the "current" CKEditor instance? (If what I just wrote makes no sense, feel free to ignore)

Reply:
Interestingly, in this case, `event.editor` property does exist, but the value is `undefined`.

I'm guessing that this property is meant to be used with events defined on editor (`editor.on(...`), which is the most common use case. But, in this case, we are defining the event on `window`. I am not sure what would even be the meaning of `event.editor`, as we are capturing a click event outside of all editors, potentially multiple editors.

I kept in the usage of `CKEDITOR.document.getWindow().on(...`, instead of using vanilla `window.addEventListener`. I am assuming this was originally used to cover some edge case, like an editor in `iframe` :thinking:

Reply:
> I'm guessing that this property is meant to be used with events defined on editor (editor.on(...), which is the most common use case. But, in this case, we are defining the event on window. I am not sure what would even be the meaning of event.editor, as we are capturing a click event outside of all editors, potentially multiple editors.

Yes, that's what I was also assuming when typing my message, but thanks for double checking @markocikos. As far as I'm concerned, since we're in "CKEDITOR land", it makes sense to use `CKEDITOR.document.getWindow().on` rather than the native `window.addEventListener` so that's all good. 

I'll approve these changes right now.